modules/actions.js

```diff
@@ -0,0 +1,104 @@
+const addTask = document.getElementById('input');
+const list = document.getElementById('tasks-list');
+
+function actions() {
+  list.innerHTML = '';
+  const tasksList = localStorage.getItem('data')
+    ? JSON.parse(localStorage.getItem('data'))
+    : [];
+  for (let i = 0; i < tasksList.length; i += 1) {
+    const input = document.createElement('input');
+    input.setAttribute('type', 'checkbox');
+    input.classList.add('checkbox');
+    input.id = i + 1;
+    if (tasksList[i].completed) {
+      input.checked = true;
+    }
+
+    const label = document.createElement('input');
+    label.value = `${tasksList[i].description}`;
+    label.classList.add('description');
+    label.setAttribute('type', 'text');
+    if (tasksList[i].completed) {
+      label.style.textDecorationLine = 'line-through';
+    }
+
+    label.setAttribute('readOnly', 'readOnly');
+    label.classList.add('description');
+
+    const edit = document.createElement('button');
+    edit.id = i + 1;
+    edit.innerHTML = '✏️';
+    edit.classList.add('edit-btn');
+
+    const deleteBtn = document.createElement('button');
+    deleteBtn.classList.add('delete-btn');
+    deleteBtn.innerHTML = '🚮';
+    deleteBtn.id = i + 1;
+
+    const li = document.createElement('li');
+    li.classList.add('container-li');
+    li.id = i + 1;
+
+    li.appendChild(input);
+    li.appendChild(label);
+    li.appendChild(edit);
+    li.appendChild(deleteBtn);
+    list.appendChild(li);
+
+    addTask.value = '';
+
+    input.addEventListener('input', (event) => {
+      if (input.checked) {
+        const idChecked = event.target.id;
+        const data = JSON.parse(localStorage.getItem('data'));
+        data[idChecked - 1].completed = true;
+        label.style.textDecorationLine = 'line-through';
+        localStorage.setItem('data', JSON.stringify(data));
+      } else if (!input.checked) {
+        const idNoChecked = event.target.id;
+        const data = JSON.parse(localStorage.getItem('data'));
+        data[idNoChecked - 1].completed = false;
+        label.style.textDecorationLine = 'none';
+        localStorage.setItem('data', JSON.stringify(data));
+      }
+      window.location.reload();
+    });
+
+    edit.addEventListener('click', (event) => {
+      if (edit.innerHTML === '✏️') {
+        label.removeAttribute('readonly');
+        label.focus();
+        edit.innerHTML = '💾';
+      } else {
+        const edited = label.value;
+        const idNumber = event.target.id;
+        label.setAttribute('readonly', 'readonly');
+        edit.innerHTML = '✏️';
+        const data = JSON.parse(localStorage.getItem('data'));
+        data[idNumber - 1].description = `${edited}`;
+        localStorage.setItem('data', JSON.stringify(data));
+      }
+    });
```

Comment:
- [ ] The user is able to add and remove tasks. Great job 👍 . However, editing the task description does not work as expected. Your code does not update the right edited task in the local storage. Kindly correct that.

<img width="734" alt="image" src="https://user-images.githubusercontent.com/77190203/193247258-7729dd17-4e96-4b02-bdde-047710edc571.png">